kaska/tests/test_s2_obs.py

```diff
@@ -43,22 +41,76 @@ def test_s2_data():
         dt.datetime(2017, 1, 7, 0, 0),
     ]
     ref_files = [
-        Path(parent+"/data/s2_data/S2A_MSIL1C_20170104.SAFE/GRANULE/IMG_DATA"),
-        Path(parent+"/data/s2_data/S2A_MSIL1C_20170107.SAFE/GRANULE/IMG_DATA"),
+        parent / "data" / "s2_data" / f"S2A_MSIL1C_{date:%Y%m%d}.SAFE" /
+        "GRANULE" / "IMG_DATA" for date in ref_dates
     ]
-    assert sorted(s2_obs.date_data) == sorted(ref_dates)
-    assert [s2_obs.date_data[d][0].parent for d in sorted(s2_obs.date_data)] == ref_files
+    assert list(s2_obs.date_data) == ref_dates
+    assert [obs[0].parent for _, obs in s2_obs.date_data.items()] == ref_files
+    print(s2_obs.parent)
 
+    # Test read_granule when all data files are available
+    output_names = ['rho_surface', 'mask', 'sza', 'vza', 'raa', 'rho_unc']
     for i, d in enumerate(ref_dates):
-        rho_surface, mask, sza, vza, raa, rho_unc = s2_obs.read_granule(d)
+        output_data = dict(zip(output_names, s2_obs.read_granule(d)))
         # np.savez_compressed(f"s2_obs_{i}_ref", rho_surface=rho_surface,
         #                    mask=mask, sza=sza, vza=vza, raa=raa,
         #                    rho_unc=rho_unc)
-        ref_data = np.load(Path(parent+f'/data/s2_obs_{i}_ref.npz'),
+        ref_data = np.load(parent / 'data' / f's2_obs_{i}_ref.npz',
                            allow_pickle=True)
-        np.testing.assert_equal(ref_data['rho_surface'], rho_surface)
-        np.testing.assert_equal(ref_data['mask'], mask)
-        np.testing.assert_equal(ref_data['sza'], sza)
-        np.testing.assert_equal(ref_data['vza'], vza)
-        np.testing.assert_equal(ref_data['raa'], raa)
-        np.testing.assert_equal(ref_data['rho_unc'], rho_unc)
+        for datum in output_names:
+            np.testing.assert_equal(ref_data[datum], output_data[datum])
+
+    # Test read_granule for missing cloud mask file
+    shutil.move(ref_files[1].parent / f"cloud.tif",
+                ref_files[1].parent / f"cloud.tif.bac")
+    output_data = dict(zip(output_names, s2_obs.read_granule(ref_dates[0])))
+    for datum in output_data:
+        assert output_data[datum] is None
```

Comment:
```python
for value in output_data.values():
    assert value is None
```
Or even better, in a single go (so we get all that are failing at once):
```python
assert list(output_data.values()) == [None] * len(output_name)
```

Reply:
Done